includes/class-wc-webhook.php

```diff
@@ -306,6 +308,26 @@ protected function is_already_processed( $arg ) {
 		return false !== array_search( $arg, $this->processed, true );
 	}
 
+	/**
+	 * Should we attempt a retry on this webhook?
+	 *
+	 * Passing in the item id and the http response to allow hooks the option of bypassing this retry.
+	 *
+	 * @since 4.4.0
+	 * @param int                    $arg Resource ID.
+	 * @param WP_Error|HTTP_Response $response HTTP Response.
+	 * @return bool
+	 */
+	public function should_retry( $arg = null, $response = null ) {
+		$webhook_retry_enabled = $this->get_retry_enabled();
+
+		// If we reached the limit of failures we won't reschedule.
+		$failed = ( $this->get_retry_count() > apply_filters( 'woocommerce_max_webhook_retry_failures', 5 ) );
+
+		$is_retryable = $webhook_retry_enabled && ! $failed;
+		return apply_filters( 'woocommerce_webhook_retry_enabled', $is_retryable, $this, $arg, $response );
```

Comment:
I think given the usage of this filter that a name like `woocommerce_webhook_retry` or `woocommerce_webhook_should_retry` might fit better, what do you think?

Reply:
Agree, `woocommerce_webhook_should_retry` is much better.